File: frontend/src/microservices/operations/airline/objects/ManagerFlightPlannerResponse.ts
// 本文件定义航空管理员航班详情响应结构。

import type { ManagerCabinInventoryPlannerResponse } from './ManagerCabinInventoryPlannerResponse'

export type ManagerFlightPlannerResponse = {
  flightId: string
  airlineId: string
  airlineName: string
  airlineCode: string
  airlineLogoPath?: string | null
  flightNumber: string
  aircraftModel?: string | null
  departureAirport: string
  arrivalAirport: string
  departureTime: string
  arrivalTime: string
  status: string
  basePrice: string
  currency: string
  createdAt: string
  cabinInventories: ManagerCabinInventoryPlannerResponse[]
}

export const managerFlightPlannerResponseFromJson = (json: string): ManagerFlightPlannerResponse =>
  JSON.parse(json) as ManagerFlightPlannerResponse

export const managerFlightPlannerResponseToJson = (value: ManagerFlightPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/airline/objects/ManagerFlightPlannerResponse.scala
// 这个文件定义 airline 航班详情响应。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ManagerFlightPlannerResponse(
    flightId: String,
    airlineId: String,
    airlineName: String,
    airlineCode: String,
    airlineLogoPath: Option[String],
    flightNumber: String,
    aircraftModel: Option[String],
    departureAirport: String,
    arrivalAirport: String,
    departureTime: String,
    arrivalTime: String,
    status: String,
    basePrice: String,
    currency: String,
    createdAt: String,
    cabinInventories: List[ManagerCabinInventoryPlannerResponse]
)
object ManagerFlightPlannerResponse:
  given sourceEncoder: Encoder[ManagerFlightPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ManagerFlightPlannerResponse] = deriveDecoder
